### PR TITLE
Bug/database resolve

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Fixes a bug where on load tsyringe isn't able to resolve the database because it has a constructor argument.

### CODE REVIEW

File: src/ioc.ts [hunk 1]
```diff
@@ -2,10 +2,12 @@ import { IocContainer } from '@tsoa/runtime'
 import { container } from 'tsyringe'
 
 import { logger, Logger, type ILogger } from './logger'
+import Database from './models/db/index.js'
 
 container.register<ILogger>(Logger, {
   useValue: logger,
 })
+container.register(Database, { useValue: new Database() })
```
Comment (n3op2) [inline src/ioc.ts:10]: would need to pay attention to this if we add a new dep to the class

**Review by n3op2 — APPROVED** [review]

File: src/ioc.ts (hunk 1 above)
Comment (mattdean-digicatapult) [reply src/ioc.ts:10]: yes

Comment (n3op2) [inline src/ioc.ts:10]:
```suggestion
container.register<IDatabase>(Database, { useValue: new Database() })
```
not sure if we have an export of typeof Database = IDatabase

Comment (mattdean-digicatapult) [reply src/ioc.ts:10]: Not necessary because this is a class. tsyringe can use the class as the specifier

**Pull request merged**